### ISSUE TITLE
submodule gdp1h() with same name in datasets and adapters

### ISSUE THREAD

**Issue by selipot** [Cloud-Drift/clouddrift#224]
Is it good practice to have submodule with same names? We both have `clouddrift.adapters.gdp1h()` and `clouddrift.datasets.gdp1h()` and I am myself getting confused.

**Comment by milancurcic** [comment]
Good question. I wasn't sure so I asked GPT4:

Is it a good practice to have two submodules in a Python package with functions that have the same names?

Having two submodules in a Python package with functions that have the same names is not inherently bad, but it can lead to confusion for developers using your package or even for yourself when maintaining it. Here's a breakdown of the pros and cons:

Pros:

Contextual Naming: If the function name makes sense in the context of each submodule, it might be logical to use the same name. For instance, a create() function in a database submodule and a create() function in a file submodule might respectively create a database record and a file.

Consistency: Consistent naming can make it easier for developers to predict function names across different submodules, leading to a smoother user experience.

Cons:

Confusion: When developers import functions from both submodules without using an alias, they will encounter a naming collision. This could lead to bugs if a developer mistakenly calls one function thinking it's the other.

Difficulty in reading code: When reading code that uses functions from these submodules, developers might need to always check the import statements to clarify which function is actually being called. This can slow down comprehension.

Maintenance issues: Developers maintaining the package might accidentally change a function in one submodule thinking it's the one in another submodule.

Documentation overhead: It might be more difficult to document functions with the same name in different submodules, especially if they have similar purposes but different implementations.

Recommendations:

Use explicit submodule naming when calling the functions: Always use the submodule's name when calling the function to make it clear which function you're referencing.

```python
from my_package import submodule1, submodule2

submodule1.my_function()
submodule2.my_function()
```

Use Aliases: If you're importing functions and there's a potential for a naming conflict, consider using an alias.

```python
from my_package.submodule1 import my_function as function1
from my_package.submodule2 import my_function as function2
```

Consider renaming: If the confusion potential is high, consider renaming one or both of the functions to better reflect their specific context or purpose.

Document thoroughly: Ensure that the package's documentation clearly explains the purposes and differences of functions with the same name in different submodules.

Use Type Annotations: Providing type hints can help in identifying which function is being used, especially in IDEs that support type hinting.

In conclusion, while it's not a strict anti-pattern to have functions with the same names in different submodules, it's crucial to consider the implications on code clarity, maintenance, and user experience. If you choose to use the same names, be deliberate and ensure that it's done in a way that minimizes confusion.

**Comment by milancurcic** [comment]
In our case, `adapters` are for developers (us) and `datasets` are for end-users. `datasets` functions can use `adapters` functions internally.

I don't think you should ever need to import both. If you do (e.g. during development or experimenting), just use the modules as namespaces, i.e. `adapters.gdp1h()` and `datasets.gdp1h`.
